### DISCUSSION TITLE
Need Help: Cara mengekstrak kata-kata dalam list dengan R

### DISCUSSION THREAD

**Discussion by nicosyuu** [teamalgoritma/community#529] [Q&A] [answered]
Selamat malam saya mau tanya saya mengambil dataset netflix dari kaggle [https://www.kaggle.com/datasets/victorsoeiro/netflix-tv-shows-and-movies?select=titles.csv](url) 
pada saat saya inspect data dengan R pada kolom genres, saya ambil contoh misalnya :	['drama', 'action', 'thriller', 'european'] ini dianggap satu kalimat oleh R
Yang saya mau tanyakan apakah R bisa membaca list itu menjadi masing masing ? 
![image](https://user-images.githubusercontent.com/103729742/216610601-4747c057-58af-44ce-b574-6bdaab47ef5b.png)
Terima kasih atas bantuannya

**Comment by onlyphantom** [comment, marked as answer]
Hi Nico, 

A few assumptions I make.
1. If there are 0 to _n_ number of potential values in `genres` and I _know_ what that `n` is, you can just do a hard separation using `,` as delimited. Let's say `n` is 4, so a movie can have up to 4 genres, I will anticipate a new dataframe with 4 new extra columns
2. Since dataframe is a data structure where each row has the same number of columns, many rows will be padded with `NA` in some of these newly created columns. A movie with exactly 4 genres will have no missing values. 

Here's the solution using `tidyr`'s `separate`. Creating `x` first:

```
x <- data.frame(runtime=c(51,114), genres=c("['documentation']", "['drama', 'action', 'european']"))
head(x)
# output:
  runtime                          genres
1      51               ['documentation']
2     114 ['drama', 'action', 'european']
```

Here's `separate` in action:
```
x <- tidyr::separate(x, genres, into=c("g1", "g2", "g3", "g4"))
# output of x is now:x
  runtime g1            g2     g3       g4
1      51    documentation            <NA>
2     114            drama action european
```

The `separate` function by default uses `[^[:alnum:]]+` as the separator, so if you are familiar with regex, you can overwrite this as well. You can modify `into` to change the names of new variables that would be created. You can also control what happens with the padding and filling. 

This uses a function from the `tidyr` package, so you need to install that first. If you prefer to do it without a package, you will have to learn some regex. For example `gsub("\\]", "", x$genres)` removes the `]` sign. We want to remove both `[` and `]` and then use a recursive loop to parse each string into new columns. A bit challenging, but doable. 

Good luck!

**Reply by nicosyuu** [reply]
Okay i'll try, thanks for your help
